File: frontend/src/app/models/request/company-request.ts
import { EBusinessType } from "../enum/ebusiness-type";
import { ECompanySize } from "../enum/ecompany-size";
import { EIndustry } from "../enum/eindustry";

export interface CompanyRequest {
    name: string;
    imageUrl?: string;
    size: ECompanySize;
    industry: EIndustry;
    businessType: EBusinessType;
    country: string;
    city: string;
    address: string;
    website?: string;
    contactsId: string[];
    salesOwnerId: string;
}

File: backend/src/main/java/org/gestionstock/stock/Payload/Request/CompanyRequest.java
package org.gestionstock.stock.Payload.Request;

import java.util.List;

import org.gestionstock.stock.Enum.EBusinessType;
import org.gestionstock.stock.Enum.ECompanySize;
import org.gestionstock.stock.Enum.EIndustry;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record CompanyRequest(
    
    @NotBlank(message = "Name is required")
    String name,
    
    String imageUrl,

    @NotNull(message = "Size is required")
    ECompanySize size,

    @NotNull(message = "Industry is required")
    EIndustry industry,
    
    @NotNull(message = "Business type is required")
    EBusinessType businessType,
    
    @NotBlank(message = "Country is required")
    String country,
    
    @NotBlank(message = "City is required")
    String city,
    
    @NotBlank(message = "Address is required")
    String address,
    
    String website,

    @NotNull(message = "Contacts id is required")
    List<String> contactsId,

    @NotBlank(message = "Sales owner id is required")
    String salesOwnerId
) {
    
}
